frontend/src/pages/games/gamesPage.js

```diff
@@ -4,141 +4,100 @@ import img2 from "../imgs/2.jpg"
-            nameAway: "team2",
-            logoHome: img1,
-            logoAway: img2,
-            scoreHome: 1,
-            scoreAway: 3,
-        },],
-
-        category2: [{
-            nameHome: "team1",
-            nameAway: "team2",
-            logoHome: img1,
-            logoAway: img2,
-            scoreHome: 1,
-            scoreAway: 3,
-        },{
-            nameHome: "team1",
-            nameAway: "team2",
-            logoHome: img1,
-            logoAway: img2,
-            scoreHome: 1,
-            scoreAway: 3,
-        },
-        {
-            nameHome: "team1",
-            nameAway: "team2",
-            logoHome: img1,
-            logoAway: img2,
-            scoreHome: 1,
-            scoreAway: 3,
-        },
-        {
-            nameHome: "team1",
-            nameAway: "team2",
-            logoHome: img1,
-            logoAway: img2,
-            time: "01/12/25"
-        },
-        {
-            nameHome: "team1",
-            nameAway: "team2",
-            logoHome: img1,
-            logoAway: img2,
-            scoreHome: 1,
-            scoreAway: 3,
-        },
-        {
-            nameHome: "team1",
-            nameAway: "team2",
-            logoHome: img1,
-            logoAway: img2,
-            scoreHome: 1,
-            scoreAway: 3,
-        },]
-    }
     const [inputValue, setInputValue] = useState('');
+    const [selectedTeam, setSelectedTeam] = useState("");
+    const [selectedDate, setSelectedDate] = useState("");
+    const [slidesCount, setSlidesCount] = useState(0);
+    const [currentPage, setCurrentPage] = useState(0);
+    const [passedGames, setPassedGames] = useState(0);
+    const gamesPerSlide = 20;
+
+    useEffect(() => {
+        let page = Math.floor(passedGames / gamesPerSlide);
+        setCurrentPage(page);
+        getGames(page);
+    }, [gamesPerSlide]);
+
+    useEffect(() => {
+        getGames(0);
+    }, []);
+
+    const applyFilters = () => {
+        getGames();
+    };
+    const [currentGames, setCurrentGames] = useState([]);
+
+    const getGames = async (page) => {
+        try {
+            const filters = {
+                team: selectedTeam,
+                date: selectedDate,
+                search: inputValue,
+            };
+
+            Object.keys(filters).forEach(
+                key => filters[key] === "" && delete filters[key]
+            );
 
-    const handleChange = (event) => {
-        setInputValue(event.target.value);
+            const response = await axios.post(
+                `${apiEndpoints.url}${apiEndpoints.games.getGames}`,
+                {
+                    pagination: {
+                        page: page + 1,
+                        per_page: gamesPerSlide,
+                    }
+                },
+                {
+                    headers: { 'Content-Type': 'application/json' },
+                }
+            );
+            setCurrentGames(response.data.items);
```

Comment:
We could use some empty lines here just to separate function calls from variable assignments